### PR TITLE
manifest annotate

### PR DESCRIPTION
**Pull request by QiWang19**
Adds manifest annotate subcommand.
Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

File: cmd/podman/manifest/annotate.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	flags.StringVar(&manifestAnnotateOpts.Variant, "variant", "", "override the `Variant` of the specified image")
+	flags.StringVar(&manifestAnnotateOpts.OSVersion, "os-version", "", "override the os `version` of the specified image")
+	flags.StringSliceVar(&manifestAnnotateOpts.Features, "features", nil, "override the `features` of the specified image")
+	flags.StringSliceVar(&manifestAnnotateOpts.OSFeatures, "os-features", nil, "override the os `features` of the specified image")
+	flags.StringSliceVar(&manifestAnnotateOpts.Annotation, "annotation", nil, "set an `annotation` for the specified image")
+}
+
+func annotate(cmd *cobra.Command, args []string) error {
+	listImageSpec := args[0]
+	instanceSpec := args[1]
```
Comment (rhatdan) [inline cmd/podman/manifest/annotate.go:43]: What happens if I give 0, 1 or 3 or more args?

File: cmd/podman/manifest/annotate.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: annotateCmd,
+		Parent:  manifestCmd,
+	})
+	flags := annotateCmd.Flags()
+	flags.StringVar(&manifestAnnotateOpts.OS, "os", "", "override the `OS` of the specified image")
```
Comment (rhatdan) [inline cmd/podman/manifest/annotate.go:32]: Options should be capatilized?

**Comment by rhatdan** [comment]
@TomSweeneyRedHat PTAL

File: cmd/podman/manifest/annotate.go (hunk 1 above)
Comment (QiWang19) [reply cmd/podman/manifest/annotate.go:43]: that won't happen. `Args:    cobra.ExactArgs(2),`

Comment (rhatdan) [reply cmd/podman/manifest/annotate.go:43]: Sadly I was looking for this.  Thanks,

Comment (TomSweeneyRedHat) [reply cmd/podman/manifest/annotate.go:43]: That is a nice option in Cobra.

File: cmd/podman/manifest/annotate.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	flags.StringVar(&manifestAnnotateOpts.Arch, "arch", "", "override the `Architecture` of the specified image")
+	flags.StringVar(&manifestAnnotateOpts.Variant, "variant", "", "override the `Variant` of the specified image")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/manifest/annotate.go:34]: Please lower case Variant and Architecture in the above lines.  I think this is what @rhatdan was asking.  I spot checked a few other podman commands, we're not uppercasing there.

File: cmd/podman/manifest/annotate.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	flags.StringVar(&manifestAnnotateOpts.OSVersion, "os-version", "", "override the os `version` of the specified image")
+	flags.StringSliceVar(&manifestAnnotateOpts.Features, "features", nil, "override the `features` of the specified image")
+	flags.StringSliceVar(&manifestAnnotateOpts.OSFeatures, "os-features", nil, "override the os `features` of the specified image")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/manifest/annotate.go:37]: Uppercase 'OS'

File: cmd/podman/manifest/annotate.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
+		Command: annotateCmd,
+		Parent:  manifestCmd,
+	})
+	flags := annotateCmd.Flags()
+	flags.StringVar(&manifestAnnotateOpts.OS, "os", "", "override the `OS` of the specified image")
+	flags.StringVar(&manifestAnnotateOpts.Arch, "arch", "", "override the `Architecture` of the specified image")
+	flags.StringVar(&manifestAnnotateOpts.Variant, "variant", "", "override the `Variant` of the specified image")
+	flags.StringVar(&manifestAnnotateOpts.OSVersion, "os-version", "", "override the os `version` of the specified image")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/manifest/annotate.go:35]: uppercase OS here too

File: cmd/podman/manifest/annotate.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	flags.StringSliceVar(&manifestAnnotateOpts.Features, "features", nil, "override the `features` of the specified image")
+	flags.StringSliceVar(&manifestAnnotateOpts.OSFeatures, "os-features", nil, "override the os `features` of the specified image")
+	flags.StringSliceVar(&manifestAnnotateOpts.Annotation, "annotation", nil, "set an `annotation` for the specified image")
+}
```
Comment (TomSweeneyRedHat) [inline cmd/podman/manifest/annotate.go:39]: To ease by OCD, could you alphabetize the above lines by field name please?  I almost missed os-version as it wasn't where I was looking for it.

File: pkg/domain/infra/abi/manifest.go [hunk 7]
```diff
@@ -100,3 +101,39 @@ func (ir *ImageEngine) ManifestAdd(ctx context.Context, opts entities.ManifestAd
 	}
 	return listID, nil
 }
+
+// ManifestAnnotate updates an entry of the manifest list
+func (ir *ImageEngine) ManifestAnnotate(ctx context.Context, names []string, opts entities.ManifestAnnotateOptions) (string, error) {
+	listImage, err := ir.Libpod.ImageRuntime().NewFromLocal(names[0])
+	if err != nil {
+		return "", errors.Wrapf(err, "error retriving local image from image name %s", names[0])
```
Comment (TomSweeneyRedHat) [inline pkg/domain/infra/abi/manifest.go:109]: retriving -> retreiving

**Comment by TomSweeneyRedHat** [comment]
A few nits here and there, but looks good overall, ty @QiWang19 !

File: pkg/domain/infra/abi/manifest.go (hunk 7 above)
Comment (giuseppe) [reply pkg/domain/infra/abi/manifest.go:109]: retreiving -> retrieving :smile:

**Pull request merged**